Extensions/TweenBehavior/tweenruntimebehavior.ts

```diff
@@ -632,30 +657,161 @@ namespace gdjs {
+                ';' +
+              Math.floor(rgbFromHslColor[1]) +
+                ';' +
+              Math.floor(rgbFromHslColor[2])
+            );
+          },
+        });
+      } else {
+        newTweenable.setConfig({
+          from: { red: fromColor[0], green: fromColor[1], blue: fromColor[2] },
+          to: { red: toColor[0], green: toColor[1], blue: toColor[2] },
+          duration: durationValue,
+          easing: easingValue,
+          step: function step(state) {
+            // @ts-ignore - objects are duck typed
+            that.owner.setColor(
+              Math.floor(state.red) +
+                ';' +
+                Math.floor(state.green) +
+                ';' +
+                Math.floor(state.blue)
+            );
+          },
+        });
+      }
+      this._addTween(
+        identifier,
+        newTweenable,
+        this._runtimeScene.getTimeManager().getTimeFromStart(),
+        durationValue
+      );
+      this._setupTweenEnding(identifier, destroyObjectWhenFinished);
+    }
+
+    /**
+     * Add an object color HSL tween, with the "to" color given using HSL (H: any number, S and L: 0-100).
+     * @param identifier Unique id to idenfify the tween
+     * @param toHue The target hue, or the same as the from color's hue if blank
+     * @param animateHue, include hue in calculations, as can't set this to -1 as default to ignore
+     * @param toSaturation The target saturation, or the same as the from color's saturation if blank
+     * @param toHue The target lightness, or the same as the from color's lightness if blank
+     * @param easingValue Type of easing
+     * @param durationValue Duration in milliseconds
+     * @param destroyObjectWhenFinished Destroy this object when the tween ends
+     */
+    
+    addObjectColorHSLTween(
+      identifier: string,
+      toHue: number,
+      animateHue: boolean,
+      toSaturation: number,
+      toLightness: number,
+      easingValue: string,
+      durationValue: float,
+      destroyObjectWhenFinished: boolean,
+    ) {
+      const that = this;
+      
+      let fromColorAsHSL : number[] = [], toColorAsHSL : number[] = [];
+      if (!this._isActive) {
+        return;
+      }
+      // @ts-ignore - objects are duck typed
+      if (!this.owner.getColor || !this.owner.setColor) {
+        return;
+      }
+      if (!!TweenRuntimeBehavior.easings[easingValue]) {
+        return;
+      }
+      
+      if (this._tweenExists(identifier)) {
+        this.removeTween(identifier);
+      }
+      
+      // @ts-ignore - objects are duck typed
+      let fromColor: any[] = this.owner.getColor().split(';');
+      fromColorAsHSL = rgbToHsl(fromColor);
+
+      let toH: number = 0, toS: number = 0, toL: number = 0;
+      if(animateHue === false || toHue.toString().trim() === "") { 
+        toH = fromColorAsHSL[0];
+      } else if(!Number.isNaN(toHue)) { // Can't check for falsy because 0 is a valid hue.
+        toH = toHue;
```

Comment:
You could use a ternary operator to do this in one line (and use const instead of let)